src/main/webapp/app/entities/participation/participation.component.ts

```diff
@@ -70,14 +88,49 @@ export class ParticipationComponent implements OnInit, OnDestroy {
                     this.isLoading = false;
                 });
                 if (this.exercise.type === ExerciseType.PROGRAMMING) {
-                    this.programmingSubmissionService.getSubmissionStateOfExercise(this.exercise.id).subscribe(() => (this.hasLoadedPendingSubmissions = true));
+                    this.programmingSubmissionService
+                        .getSubmissionStateOfExercise(this.exercise.id)
+                        .pipe(
+                            tap((exerciseSubmissionState: ExerciseSubmissionState) => {
+                                this.exerciseSubmissionState = exerciseSubmissionState;
+                            }),
+                        )
+                        .subscribe(() => (this.hasLoadedPendingSubmissions = true));
                 }
                 this.newManualResultAllowed = areManualResultsAllowed(this.exercise);
                 this.presentationScoreEnabled = this.checkPresentationScoreConfig();
             });
         });
     }
 
+    updateParticipationFilter(newValue: FilterProp) {
+        this.isLoading = true;
+        setTimeout(() => {
+            this.participationCriteria.filterProp = newValue;
+            this.isLoading = false;
+        });
+    }
+
+    filterParticipationByProp = (participation: Participation) => {
+        switch (this.participationCriteria.filterProp) {
+            case FilterProp.FAILED:
+                return this.hasFailedSubmission(participation);
+            case FilterProp.ALL:
+                return true;
+            default:
+                return true;
+        }
+    };
+
+    hasFailedSubmission(participation: Participation) {
```

Comment:
Can you make it private since it is only used inside this class?

Reply:
Good catch, done.